lib/ts/nextjs/middleware.ts

```diff
@@ -0,0 +1,438 @@
+    const response = new Response(null, {});
+    response.headers.set("x-middleware-next", "1");
+    return response;
+}
+
+async function fetchNewTokens(request: Request): Promise<{
+    headers: [string, string][];
+    cookies: string[];
+}> {
+    const refreshApiURL = new URL(
+        `${AppInfo.apiBasePath.getAsStringDangerous()}/session/refresh`,
+        AppInfo.apiDomain.getAsStringDangerous()
+    );
+    const cookieRefreshToken = getCookie(request, REFRESH_TOKEN_COOKIE_NAME);
+    const headerRefreshToken = getCookie(request, REFRESH_TOKEN_HEADER_NAME);
+    const refreshRequestHeaders = new Headers();
+    refreshRequestHeaders.append("Content-Type", "application/json");
+    if (cookieRefreshToken) {
+        refreshRequestHeaders.append("Cookie", `${REFRESH_TOKEN_COOKIE_NAME}=${cookieRefreshToken}`);
+    } else if (headerRefreshToken) {
+        refreshRequestHeaders.append("Authorization", `Bearer ${headerRefreshToken}`);
+    }
+
+    const refreshResponse = await fetch(refreshApiURL, {
+        method: "POST",
+        headers: refreshRequestHeaders,
+        credentials: "include",
+    });
+
+    if (!refreshResponse.ok) {
+        throw new Error(`Refresh request returned an invalid status code: ${refreshResponse.status}`);
+    }
+
+    logDebugMessage("Session refresh request completed");
+    const cookies: string[] = [];
+    const tokens = {
+        accessToken: refreshResponse.headers.get(ACCESS_TOKEN_HEADER_NAME) || "",
+        refreshToken: refreshResponse.headers.get(REFRESH_TOKEN_HEADER_NAME) || "",
+        frontToken: refreshResponse.headers.get(FRONT_TOKEN_HEADER_NAME) || "",
+        antiCsrfToken: refreshResponse.headers.get(ANTI_CSRF_TOKEN_HEADER_NAME) || "",
+    };
+
+    cookies.push(`${FRONT_TOKEN_COOKIE_NAME}=${tokens.frontToken}; Path=/`);
+    // TOOD: Review the current build target
+    // getSetCookie was added in node 18 and our build target is ES5
+    // This should not a problem here since the function runs in the Vercel edge runtime environment
+    // @ts-expect-error TS(2339): Property 'getSetCookie' does not exist on type 'Headers'.
+    const setCookieHeaders = refreshResponse.headers.getSetCookie();
+    for (const header of setCookieHeaders) {
+        if (header.includes(ACCESS_TOKEN_COOKIE_NAME)) {
+            cookies.push(header);
+        }
+        if (header.includes(REFRESH_TOKEN_COOKIE_NAME)) {
+            cookies.push(header);
+        }
+        if (header.includes(ANTI_CSRF_TOKEN_COOKIE_NAME)) {
+            cookies.push(header);
+        }
+    }
+
+    if (!cookies.some((cookie) => cookie.includes(ACCESS_TOKEN_COOKIE_NAME))) {
+        cookies.push(`${ACCESS_TOKEN_HEADER_NAME}=${tokens.accessToken}; Path=/; HttpOnly; SameSite=Lax`);
+    }
+    if (!cookies.some((cookie) => cookie.includes(REFRESH_TOKEN_COOKIE_NAME))) {
+        cookies.push(
+            `${REFRESH_TOKEN_HEADER_NAME}=${tokens.refreshToken}; Path=/api/auth/session/refresh; HttpOnly; SameSite=Lax`
```

Comment:
I'm not sure this `Path` is correct. This is intended to be a frontend cookie; I believe it's set without a path.

Reply:
As far as I can tell, from the inspecting existing refresh cookies and from what [the node sdk](https://github.com/supertokens/supertokens-node/blob/master/lib/ts/recipe/session/cookieAndHeaders.ts#L175) sends to the client a separate path is set on the cookie.

A potential bug here is that I'm not taking into account the AppInfo when setting the path. I fixed with the latest change.